spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -1641,6 +1652,17 @@ else if (offset >= 0) {
 			}
 		}
 
+		@Override
+		public void seekToTimestamp(String topic, int partition, long timestamp) {
+			this.seeks.add(new TopicPartitionOffset(topic, partition, timestamp, SeekPosition.TIMESTAMP));
+		}
+
+		@Override
+		public void seekToTimestamp(Collection<TopicPartition> topicParts, long timestamp) {
+			this.seeks.addAll(topicParts.stream()
```

Comment:
And why can't we do a simple iteration with a plain delegation to the `seekToTimestamp(String topic, int partition, long timestamp)`? I'm really not a fun of Java Stream at runtime. It's OK to use them during initialization, but if the code is called several times at runtime, it would be better do not use them in the Framework.